Code Challenge

YourWishes opened the pull request:
Completed the required challenges, any issues feel free to contact me!

danwfeeney commented on test-one/package.json:
```diff
@@ -7,7 +7,7 @@
     "test": "./node_modules/.bin/_mocha \"lib/**/*-tests.js\"",
     "go": "node lib/index",
-    "start": "nodemon -x \"npm run compile && npm run test && npm go\""
+    "start": "nodemon -x \"npm run compile && npm run test && npm run go\""
```
good spot, thanks for picking this up, I'e fixed in the master repo now

danwfeeney commented on test-one/src/bosshog.js:
```diff
@@ -0,0 +1,33 @@
+
+let logger = new Logger();
+
+class Bosshog {
```
👍  for putting this into it's own class

danwfeeney commented on test-one/src/bosshog.js:
```diff
@@ -0,0 +1,33 @@
+
+    }
+
+    print(min, max) {
```
is there a way we could write this to make it more reusable and testable?

JiEunYou replied:
I created getResult function and returned an array for re-usability and test-ability.

danwfeeney commented on test-one/src/bosshog.js:
```diff
@@ -0,0 +1,33 @@
+    print(min, max) {
+        for (var i = min; i <= max; i++) {
+
+            if (i % 3 === 0 && i % 5 === 0) {
```
would be nice to have some unit tests around this functionality

JiEunYou replied:
I added one :)

danwfeeney commented on test-one/src/bosshog.js:
```diff
@@ -0,0 +1,33 @@
+                continue;
+            }
+
+            logger.log(i);
```
should we be mixing display logic with business logic?

JiEunYou replied:
Great point. separation of concerns. Now I return an array.

danwfeeney commented on test-two/src/services/github.js:
```diff
@@ -1,16 +1,48 @@
+    paging (queryPromise) {
+        var arr = [];
+
+        var loopingPromise = function (page) {
```
shouldn't we be using ES6 syntax for declarations and functions to be consistent with the existing code base?

JiEunYou replied:
I rewrote this part using =>

danwfeeney commented on test-two/src/services/github.js:
```diff
@@ -1,16 +1,48 @@
         }
     }
 
+    paging (queryPromise) {
```
is there a faster way to return all of the pages?

JiEunYou replied:
It might be possible to extract the number of pages from header. Yet it seems have a minor impact compared to the total number of requests. The slow part is getting all repositories.

danwfeeney commented on test-two/src/services/github.js:
```diff
@@ -1,16 +1,48 @@
+        return loop(1);
+    }
+
     getUsersForOrganisation(organisationId) {
```
Is there a way to have written this class to be more reusable? It's not always going to be practical to return all users.

JiEunYou replied:
Maybe we could add page parameter to get a specific page.

danwfeeney commented on test-two/src/services/github-tests.js:
```diff
@@ -81,4 +81,46 @@ describe("github service", () => {
         httpGetStub.getCall(0).args[0].startsWith(baseGitHubUrl).should.equal(true);
     });
+
+     it("should return repositories for user", (done) => {
```
so nice to see unit tests around this stuff 👍

JiEunYou replied:
Thank you :)

danwfeeney commented on test-two/src/services/github.js:
```diff
@@ -1,16 +1,48 @@
+import PromiseLoop from "promise-loop";
```
Doesn't look like this dependency has been added to the package.json file for this project. Once I added it the solution worked nicely.

danwfeeney replied:
Do we really need this package? Can you explain what it's doing?

JiEunYou replied:
I have tried to remove it and used chaining promises. It took some trial and errors :)

JiEunYou replied:
The dependency has been removed now.

YourWishes committed:
Updated test-one

Added program to print numbers 1 through to 100, with case checking for certain conditions.

YourWishes committed:
Finished step-3

Added a basic react website, some simple menu and a fun generic logo.

Moved menu-data.json to a better subdirectory

Webpack and SCSS supported compilation, no testing required as of now.

Code is a little messy, could do with a few revisions of cleanup.

YourWishes committed:
Fixed Tab Inconsistencies

This is what I deserve for using a windows system.
